mitiq/lre/tests/test_multivariate_richardson.py

```diff
@@ -0,0 +1,221 @@
+)
+test_circuit2 = Circuit(
+    [ops.H.on_each(*qreg1)],
+    [ops.CNOT.on(qreg1[0], qreg1[1])],
+    [ops.X.on(qreg1[2])],
+    [ops.TOFFOLI.on(*qreg1)],
+)
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree, expected",
+    [
+        (1, 1, [(0,), (1,)]),
+        (
+            2,
+            2,
+            [(0, 0), (1, 0), (0, 1), (2, 0), (1, 1), (0, 2)],
+        ),
+        (
+            3,
+            2,
+            [
+                (0, 0, 0),
+                (1, 0, 0),
+                (0, 1, 0),
+                (0, 0, 1),
+                (2, 0, 0),
+                (1, 1, 0),
+                (0, 2, 0),
+                (1, 0, 1),
+                (0, 1, 1),
+                (0, 0, 2),
+            ],
+        ),
+    ],
+)
+def test_basis_exp(test_num_layers, test_degree, expected):
+    assert (
+        _full_monomial_basis_term_exponents(test_num_layers, test_degree)
+        == expected
+    )
+
+
+@pytest.mark.parametrize(
+    "test_num_layers, test_degree",
+    [(1, 1), (2, 2), (3, 2), (10, 4)],
+    # Note need to add (100, 2) here. This makes the unit test very slow.
+)
+def test_basis_exp_len(test_num_layers, test_degree):
+    calc_dict = _full_monomial_basis_term_exponents(
+        test_num_layers, test_degree
+    )
+    for i in calc_dict:
+        assert len(i) == test_num_layers
+
+
+@pytest.mark.parametrize(
+    "test_circ, test_degree, expected_matrix",
+    [
+        (
+            test_circuit1,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 3.0, 1.0, 9.0, 3.0, 1.0],
+                    [1.0, 1.0, 3.0, 1.0, 3.0, 9.0],
+                    [1.0, 5.0, 1.0, 25.0, 5.0, 1.0],
+                    [1.0, 3.0, 3.0, 9.0, 9.0, 9.0],
+                    [1.0, 1.0, 5.0, 1.0, 5.0, 25.0],
+                ]
+            ),
+        ),
+        (
+            test_circuit2,
+            2,
+            np.array(
+                [
+                    [1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0, 1.0],
+                    [1.0, 3.0, 1.0, 1.0, 9.0, 3.0, 1.0, 3.0, 1.0, 1.0],
+                    [1.0, 1.0, 3.0, 1.0, 1.0, 3.0, 9.0, 1.0, 3.0, 1.0],
+                    [1.0, 1.0, 1.0, 3.0, 1.0, 1.0, 1.0, 3.0, 3.0, 9.0],
+                    [1.0, 5.0, 1.0, 1.0, 25.0, 5.0, 1.0, 5.0, 1.0, 1.0],
+                    [1.0, 3.0, 3.0, 1.0, 9.0, 9.0, 9.0, 3.0, 3.0, 1.0],
+                    [1.0, 3.0, 1.0, 3.0, 9.0, 3.0, 1.0, 9.0, 3.0, 9.0],
+                    [1.0, 1.0, 5.0, 1.0, 1.0, 5.0, 25.0, 1.0, 5.0, 1.0],
+                    [1.0, 1.0, 3.0, 3.0, 1.0, 3.0, 9.0, 3.0, 9.0, 9.0],
+                    [1.0, 1.0, 1.0, 5.0, 1.0, 1.0, 1.0, 5.0, 5.0, 25.0],
+                ]
+            ),
+        ),
+    ],
+)
+def test_sample_matrix(test_circ, test_degree, expected_matrix):
+    assert (
+        expected_matrix - sample_matrix(test_circ, test_degree, 1)
+    ).all() <= 1e-3
```

Comment:
Thanks for catching this! 

Apparently, the two failures I noticed in https://github.com/unitaryfund/mitiq/pull/2447#issuecomment-2293634688 were related to this.  

Because the tests passed, I thought there was a float addition error. I will work on finding a way around the `for` block where I was copying the array.